massa-protocol-worker/src/handlers/block_handler/retrieval.rs

```diff
@@ -463,6 +463,23 @@ impl RetrievalThread {
                 received: header.content.current_version,
             })
         } else {
+            // announced version == 0 -> No announce
+            if header.content.announced_version != 0 {
+                if header.content.announced_version <= version {
+                    // Received block, announced an old version (already accepted)
+                    return Err(ProtocolError::InvalidAnnouncedNetworkVersion {
+                        local: version,
+                        announced_received: header.content.announced_version,
+                    });
+                }
+
+                let version_to_announce = self.mip_store.get_network_version_to_announce();
+                if header.content.announced_version != version_to_announce {
+                    warn!("Please update your node if you want to support this update (network version: {})", 
+                        header.content.announced_version);
+                }
```

Comment:
Not a fan of putting out a warning for version support in the logs of the node that receives an outdated header...

Reply:
But here you're using `!=` which means that the receiving node can be up to date, receive an outdated header, and still ask the up to date node to upgrade for support. What I'm saying is have a `header.content.announced_version > version_to_announce` check, and for the case where `header.content.announced_version < version_to_announce` I imagine the creator node will eventually get the warning when an up to date node sends him a block.

Reply:
fixed

Reply:
One more comment: we must not tell the node runner to update their node every time they receive a block announcing something unknown. The warning should only appear following the counters: eg. if more than 30% of the network is announcing something we don't know, then warn the user. Otherwise attackers can cause a lot of noise